File: trading-ui/jgod-trading-ui/src/types/observer.ts
/**
 * Knowledge Brain Observer Types
 * 
 * TypeScript type definitions for Observer API.
 */

export type KnowledgeGovernanceSummary = {
  timestamp: string; // ISO datetime string
  total_sections: number;
  pending_review_count: number;
  critical_alerts_active: number;
  sections_modified_last_7d: number;
  simulations_last_30d: number;
  sim_approve_rate_30d: number;
  sim_maxdd_increase_rate_30d: number;
  s_rank_recalculations_last_24h: number;
  s_rank_strategy_degradation_7d: number;
  s_rank_distribution: {
    S: number;
    A: number;
    B: number;
    C: number;
    D: number;
  };
};

export type StabilityAlert = {
  severity: "CRITICAL" | "WARNING" | "INFO";
  message: string;
  timestamp: string; // ISO datetime string
};

export type SRankDistributionHistory = {
  date: string; // ISO date string
  distribution: {
    S: number;
    A: number;
    B: number;
    C: number;
    D: number;
  };
};

File: jgod/api/schemas/observer.py
"""Knowledge Brain Observer API Schemas

Pydantic models for Observer API endpoints.
"""

from datetime import datetime
from typing import Dict, List
from pydantic import BaseModel


class KnowledgeGovernanceSummarySchema(BaseModel):
    """知識治理狀態概覽 Schema"""
    timestamp: datetime
    total_sections: int = 0
    pending_review_count: int = 0
    critical_alerts_active: int = 0
    sections_modified_last_7d: int = 0
    simulations_last_30d: int = 0
    sim_approve_rate_30d: float = 0.0
    sim_maxdd_increase_rate_30d: float = 0.0
    s_rank_recalculations_last_24h: int = 0
    s_rank_strategy_degradation_7d: int = 0
    s_rank_distribution: Dict[str, int] = {}


class StabilityAlertSchema(BaseModel):
    """穩定性警報 Schema"""
    severity: str  # "CRITICAL" | "WARNING" | "INFO"
    message: str
    timestamp: datetime


class SRankDistributionHistorySchema(BaseModel):
    """S-Rank 分佈歷史數據 Schema"""
    date: str  # ISO date string
    distribution: Dict[str, int]  # S/A/B/C/D counts

